launch/launch/actions/execute_process.py

```diff
@@ -513,6 +519,23 @@ def __expand_substitutions(self, context):
             self.__logger.info("process details: cmd=[{}], cwd='{}', custom_env?={}".format(
                 ', '.join(cmd), cwd, 'True' if env is not None else 'False'
             ))
+        try:
+            emulate_tty = yaml.safe_load(
+                context.launch_configurations['emulate_tty']
+            )
+            if type(emulate_tty) is not bool:
+                raise TypeError('emulate_tty is not boolean [{}]'.format(
+                    type(emulate_tty)
+                ))
+        except KeyError:
+            emulate_tty = self.__emulate_tty
+        # guard against windows - tty emulation is not yet working
+        #     https://github.com/ros2/launch/issues/268
+        if emulate_tty and platform.system() == 'Windows':
+            self.__logger.warning(
+                'tty emulation not yet supported on windows, disabling'
+            )
+            emulate_tty = False
```

Comment:
Is this needed? Doesn't setting `emulate_tty` to `True` on Windows gracefully degrade?

Reply:
I wanted to ensure there was a logged warning that doesn't become contradictory if, for whatever reason, it started gracefully working. This made sure that it never provides an incorrect message.

I expect the better solution would be to explicitly test for the capability of tty emulation pre-execution, or to check for tty emulation post-execution and issuing a warning based on that. I'm not exactly certain how to implement that right now though.

Reply:
That's what I'm saying, `osrf_pycommon` already tests this, e.g. if `import pty` fails it will automatically fallback to non-emulation:

https://github.com/osrf/osrf_pycommon/blob/8ba78744d9df7de70d824b4d76c11afe9e81ccd1/osrf_pycommon/process_utils/impl.py#L154-L156

So I think the warning is ok, though I'm not convinced we'll ever support it and it might just be annoying to Windows users who will otherwise have to conditionally set this setting. But either way, I don't think you need to set `emulate_tty` to `False` here.

The only reason it was failing before was that your test asserted that emulation was happening, and on Windows it wasn't. With the work around in the test, this check and logic and warning I think are unnecessary. Though we could keep the warning if we need to.

It will also gracefully degrade if you run out of pty's which can happen.

Reply:
Sorry for the latency - dropped in the last commit. That there (osrf_pycommon) is certainly doing the guarding and if a warning is desired, probably the best place for it so not not going to make any decision here.